presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaPageSourceProvider.java

```diff
@@ -124,17 +129,19 @@ private InternalTable getInternalTable(ConnectorTransactionHandle transactionHan
         InformationSchemaTableHandle handle = split.getTableHandle();
         Map<String, NullableValue> filters = split.getFilters();
 
+        Identity identity = connectorSession.getIdentity();
         Session session = Session.builder(metadata.getSessionPropertyManager())
-                .setTransactionId(transaction.getTransactionId())
                 .setQueryId(new QueryId(connectorSession.getQueryId()))
-                .setIdentity(connectorSession.getIdentity())
+                .setIdentity(new Identity(identity.getUser(), identity.getPrincipal()))
```

Comment:
This is kind of hack. InformationSchema connector receives `ConnectorSession`, but to be able to call Metadata methods it needs `Session`. So, Here we are re-creating session based on the connector session. And the identity in `Session` is not supposed to have any role assigned. Because we consider roles only on the connector level.